src/Development/IDE/Core/PositionMapping.hs

```diff
@@ -46,19 +46,23 @@ toCurrent (Range (Position startLine startColumn) (Position endLine endColumn))
     | line > endLine || line == endLine && column >= endColumn =
       -- Position is after the change so increase line and column number
       -- as necessary.
-      Just $ Position (line + lineDiff) newColumn
+      Just $ newLine `seq` newColumn `seq` Position newLine newColumn
     | otherwise = Nothing
     -- Position is in the region that was changed.
     where
         lineDiff = linesNew - linesOld
         linesNew = T.count "\n" t
-        linesOld = endLine - startLine
+        linesOld = fromIntegral endLine - fromIntegral startLine
+        newEndColumn :: UInt
         newEndColumn
-          | linesNew == 0 = startColumn + T.length t
-          | otherwise = T.length $ T.takeWhileEnd (/= '\n') t
+          | linesNew == 0 = fromIntegral $ fromIntegral startColumn + T.length t
+          | otherwise = fromIntegral $ T.length $ T.takeWhileEnd (/= '\n') t
+        newColumn :: UInt
         newColumn
-          | line == endLine = column + newEndColumn - endColumn
+          | line == endLine = fromIntegral $ (fromIntegral column + newEndColumn) - fromIntegral endColumn
```

Comment:
why do we need any conversions here? It looks like all of these are `UInt`s which has a `Num` instance.

Reply:
That's a good point. I worried that `UInt`'s modular arithmetic (it's a newtype over [`Data.Mod.Word.Mod (2^31)`](https://hackage.haskell.org/package/mod-0.1.2.2/docs/Data-Mod-Word.html#t:Mod)) might be a problem, but the code around already assumes that `startX` <= `endX`, so the conversions shouldn't be necessary